compiler/qsc_parse/src/prim.rs

```diff
@@ -170,12 +170,29 @@ pub(super) fn many<T>(s: &mut Scanner, mut p: impl Parser<T>) -> Result<Vec<T>>
     Ok(xs)
 }
 
-pub(super) fn seq<T>(s: &mut Scanner, mut p: impl Parser<T>) -> Result<(Vec<T>, FinalSep)> {
+pub(super) fn seq<T>(s: &mut Scanner, mut p: impl Parser<T>) -> Result<(Vec<T>, FinalSep)>
```

Comment:
Okay, every time I see an iteration of this fix, I ask for more, but...

Would it be possible to also recover from broken syntax, not just _missing_ entries? Like in `foo( , . , -- ,)` . Can `.` and `--` be turned into error expression nodes?

The scenario I have in mind is you start typing `DrawRandomInt(-)` , starting to type a negative integer literal, and signature help disappears.

I feel like this _could_ be achievable by using the recovering parser and `,` as a barrier token. In my experiments, I did manage to do it for expression sequences, but failed to generalize it for other kinds of sequences.

Reply:
I now see that Scott already pointed to an issue he filed on this above.

Reply:
Yeah, I think these are great suggestions for us to continue down this road, and would prefer to address them in a follow up PR focused on expr recovery.